### PR TITLE
upgrade dependencies

### PR DESCRIPTION
**Pull request by st-h**
- upgrade all dependencies to latest version
- fix sinon.stub deprecation
- replace deprecated ember merge() with assign()

### CODE REVIEW

**Comment by joshsmith** [comment]
@st-h would you be willing to re-update deps here? I'm also curious as to why coverage decreased so much.

**Comment by st-h** [comment]
@joshsmith Sure, I can re-update this PR. As for the coverage, I had a look back then, but could not find anything troublesome. I had a look again today and if you look at builds with high coverage like https://coveralls.io/builds/22709247 it lists only the `app` and `tests` folder.
If you look at the coverage of this PR https://coveralls.io/builds/20926446 it interestingly lists the `addon` folder as well. The coverage of the `app` folder is identical to previous builds. My guess into the wild was that updating dependencies lead to a more accurate coverage report - however, so far I never found an occasion to really look into this and confirm my assumption.

**Comment by joshsmith** [comment]
Awesome, thanks. As soon as this is ready for-review and you accept my invite, just request a review from me and I'll merge in. Then I can cut a new release and push to npm.

**Comment by st-h** [comment]
@joshsmith I think this should be good to go now. I am just wondering why the last commit did not trigger a new travis run... 🤔 

ah, travis complains: `Branch "develop" not included per configuration` not sure why I am seeing this just now

File: .travis.yml [hunk 1]
```diff
@@ -18,7 +18,28 @@ env:
   global:
     # See https://git.io/vdao3 for details.
     - JOBS=1
-  matrix:
+
+branches:
+  only:
+    - master
+    # npm version tags
+    - /^v\d+\.\d+\.\d+/
```
Comment (joshsmith) [inline .travis.yml]: These lines are what is preventing the build from running now. It's only running against `master` and npm version tags now. Can we also run against `develop`? Or just run everywhere by default?

**Review by joshsmith — CHANGES_REQUESTED** [review]

**Comment by joshsmith** [comment]
@st-h build is now failing due to the `ember-try` config. Looks like the changes have resulted in some mangled objects.

File: .travis.yml (hunk 1 above)
Comment (st-h) [reply .travis.yml]: thanks for taking a look. Didn't think about that this might be related to the .travis config :facepalm's

**Comment by st-h** [comment]
@joshsmith got it. the upgrade added linting to the travis file, however it is missing within package.json. I'll push some fixes.

**Comment by st-h** [comment]
@joshsmith phew, that took longer than I expected... and TIL: just the presence of old bower files can fail your build.

**Comment by st-h** [comment]
I also disabled most of the linting as this branch contains deprecations that were fixed in other PRs. And I just can't merge my master as it contains all the changes to publish under a different name. We should look into reenabling linting after this was merged and see what remains to be fixed.

**Review by joshsmith — APPROVED** [review]
Excellent, thanks for the work on this! I'll merge and then you can feel free to PR the linting change.

**Comment by danwenzel** [comment]
Hi @joshsmith - Will there be a release published for the recent work on this repo?

**Comment by danwenzel** [comment]
Hey @joshsmith - Trying again... We're doing some work with stripe elements for the upcoming SCA requirements, and we'd love to use the latest version of this addon.

I've been testing it out with the latest commit of the `develop` branch, and so far it's worked perfectly!

Can there be a new version published to npm?

**Comment by lindyhopchris** [comment]
@joshsmith should we get a `v0.5.0` tagged from `develop`? might be useful and then work out what we need to do next to get this towards `v1.0`?

**Comment by josemarluedke** [comment]
@joshsmith @snewcomer Can we get a release out containing these changes?

**Pull request merged**